core/packages/contracts/src/BeefyClient.sol

```diff
@@ -520,41 +533,44 @@ contract BeefyClient is Ownable {
      * the previous version did not require any sorting.
      *
      * @dev Checks if a validators address is a member of the merkle tree
-     * @param addr The address of the validator to check
+     * @param account The address of the validator to check
      * @param proof Merkle proof required for validation of the address
      * @return true if the validator is in the set
      */
-    function isValidatorInSet(ValidatorSet memory vset, address addr, uint256 index, bytes32[] calldata proof)
+    function isValidatorInSet(ValidatorSet memory vset, address account, uint256 index, bytes32[] calldata proof)
         internal
         pure
         returns (bool)
     {
-        bytes32 hashedLeaf = keccak256(abi.encodePacked(addr));
+        bytes32 hashedLeaf = keccak256(abi.encodePacked(account));
         return MerkleProof.verify(vset.root, hashedLeaf, index, vset.length, proof);
     }
 
-    /**
-     * @dev Helper to create an initial validator bitfield.
-     */
-    function createInitialBitfield(uint256[] calldata bitsToSet, uint256 length)
-        external
-        pure
-        returns (uint256[] memory)
-    {
-        return Bitfield.createBitfield(bitsToSet, length);
-    }
-
-    /**
-     * @dev Helper to create a final bitfield, with subsampled validator selections
-     */
-    function createFinalBitfield(bytes32 commitmentHash, uint256[] calldata bitfield)
-        external
+    // Basic checks for commitment
+    function validate(Commitment calldata commitment, uint256[] calldata bitfield)
+        internal
         view
-        returns (uint256[] memory)
+        returns (bytes32, bytes32)
     {
-        Ticket storage ticket = tickets[createTicketID(msg.sender, commitmentHash)];
-        return Bitfield.subsample(
-            ticket.prevRandao, bitfield, minimumSignatureThreshold(ticket.validatorSetLen), ticket.validatorSetLen
-        );
+        bytes32 commitmentHash = keccak256(encodeCommitment(commitment));
+        bytes32 ticketID = createTicketID(msg.sender, commitmentHash);
+        Ticket storage ticket = tickets[ticketID];
+
+        if (ticket.blockNumber == 0) {
+            revert InvalidTicket();
```

Comment:
Actually this is checking whether the ticket actually exists at all. Since Solidity does not have a `null` construct, `tickets[ticketID]` will return a struct with all fields initialised to zero.

Reply:
I would assume beefy in production network(i.e. kusama/polkadot) will be enabled from some future block and not start from genesis. `ticket.blockNumber == 0`  is the case that we can't find a matched `commitHash` when `submitFinal*`, there is a [test case](https://github.com/Snowfork/snowbridge/blob/362e0eabe9bf90aaa2f145e32fa2da3552059856/core/packages/contracts/test/BeefyClient.t.sol#L343-L355) specific for it. 

Previously it will error as `PrevRandaoNotCaptured` but definitely not the case. I would prefer we make error more clear and accurate.